internal/services/securitycenter/security_center_server_vulnerability_assessment_hybrid_vm_resource.go

```diff
@@ -0,0 +1,146 @@
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/security/mgmt/v3.0/security"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/tf"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	computeParse "github.com/hashicorp/terraform-provider-azurerm/internal/services/compute/parse"
+	computeValidate "github.com/hashicorp/terraform-provider-azurerm/internal/services/compute/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+const (
+	hybridProvider = "Microsoft.HybridCompute"
+	hybridType     = "machines"
+)
+
+func resourceServerVulnerabilityAssessmentHybridVM() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceServerVulnerabilityAssessmentHybridVMCreate,
+		Read:   resourceServerVulnerabilityAssessmentHybridVMRead,
+		Delete: resourceServerVulnerabilityAssessmentHybridVMDelete,
+
+		// TODO: replace this with an importer which validates the ID during import
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(5 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(10 * time.Minute),
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+			"hybrid_machine_id": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: computeValidate.HybridMachineID,
+			},
+		},
+	}
+}
+
+func resourceServerVulnerabilityAssessmentHybridVMCreate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).SecurityCenter.ServerVulnerabilityAssessmentClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	hybridMachineId, err := computeParse.HybridMachineID(d.Get("hybrid_machine_id").(string))
+	if err != nil {
+		return err
+	}
+
+	vulnerabilityAssessment, err := client.Get(ctx, hybridMachineId.ResourceGroup, hybridProvider, hybridType, hybridMachineId.MachineName)
+	if err != nil {
+		if !utils.ResponseWasNotFound(vulnerabilityAssessment.Response) {
+			return fmt.Errorf("checking for presence of existing Advanced Threat Protection for %s: %+v", *hybridMachineId, err)
+		}
+	}
+
+	if vulnerabilityAssessment.ID != nil && *vulnerabilityAssessment.ID != "" {
+		return tf.ImportAsExistsError("azurerm_security_center_server_vulnerability_assessment_hybrid_vm", *vulnerabilityAssessment.ID)
+	}
+
+	vulnerabilityAssessment, err = client.CreateOrUpdate(ctx, hybridMachineId.ResourceGroup, hybridProvider, hybridType, hybridMachineId.MachineName)
+	if err != nil {
+		return fmt.Errorf("create Server Vulnerability Assessment for %s: %+v", *hybridMachineId, err)
+	}
+
+	if vulnerabilityAssessment.ID != nil {
+		d.SetId(*vulnerabilityAssessment.ID)
+	}
```

Comment:
rather than using the Resource ID returned from the API here can we instead define a Resource ID (`resourceids.go`) and use that here instead?